include/zephyr/net/http/server.h

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2023, Emna Rekik
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_HTTP_SERVER_H_
+#define ZEPHYR_INCLUDE_NET_HTTP_SERVER_H_
+
+#include <stdint.h>
+
+#if defined(CONFIG_POSIX_API)
+
+#include <stddef.h>
+#include <poll.h>
+
+#else
+
+#include <zephyr/posix/poll.h>
+
+#endif
+
+#include <zephyr/net/http/parser.h>
+
+#define CLIENT_BUFFER_SIZE         CONFIG_NET_HTTP_SERVER_CLIENT_BUFFER_SIZE
+#define MAX_SERVICES               CONFIG_NUM_HTTP_SERVICES
+#define MAX_CLIENTS                CONFIG_NET_HTTP_SERVER_MAX_CLIENTS
+#define MAX_STREAMS                CONFIG_NET_HTTP_SERVER_MAX_STREAMS
+
+enum http_support_method {
```

Comment:
This one too - we already have `http/method.h`, and applying BIT(x) at compile time or runtime is relatively trivial.

I believe one or more of the official methods have been deprecated, so likely all possible supported bits *could* fit into a 32-bit int.

It also doesn't make sense to me to skip using `BIT(0)` here...

Reply:
The values from method.h (`enum http_method`) are values 0-32. Currently http_support_method is used as a bit field where multiple methods can be set in one variable. So we cannot use the values from method.h directly.

Reply:
> The values from method.h (`enum http_method`) are values 0-32. Currently http_support_method is used as a bit field where multiple methods can be set in one variable. So we cannot use the values from method.h directly.

Sure we can - just `BIT(a) | BIT(b)`

Reply:
Sure, I just was wondering how to present this to the user. Should the user do the BIT() calls in the application or should we provide ready made defines for it.